src/libs/actions/connections/QuickbooksOnline.ts

```diff
@@ -86,6 +86,87 @@ function buildOnyxDataForMultipleQuickbooksConfigurations<TConfigUpdate extends
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                connections: {
+                    [CONST.POLICY.CONNECTIONS.NAME.QBO]: {
+                        config: {
+                            autoSync: {
+                                enabled: settingValue ?? null,
+                            },
+                            pendingFields: {
+                                [CONST.QUICKBOOKS_CONFIG.AUTO_SYNC]: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+                            },
+                            errorFields: {
+                                [CONST.QUICKBOOKS_CONFIG.AUTO_SYNC]: null,
+                            },
+                        },
+                    },
+                },
+            },
+        },
+    ];
+
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                connections: {
+                    [CONST.POLICY.CONNECTIONS.NAME.QBO]: {
+                        config: {
+                            autoSync: {
+                                enabled: !settingValue ?? null,
+                            },
+                            pendingFields: {
+                                [CONST.QUICKBOOKS_CONFIG.AUTO_SYNC]: null,
+                            },
+                            errorFields: {
+                                [CONST.QUICKBOOKS_CONFIG.AUTO_SYNC]: ErrorUtils.getMicroSecondOnyxErrorWithTranslationKey('common.genericErrorMessage'),
+                            },
+                        },
+                    },
+                },
+            },
+        },
+    ];
+
+    const successData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                connections: {
+                    [CONST.POLICY.CONNECTIONS.NAME.QBO]: {
+                        config: {
+                            autoSync: {
+                                enabled: settingValue ?? null,
+                            },
+                            pendingFields: {
+                                [CONST.QUICKBOOKS_CONFIG.AUTO_SYNC]: null,
+                            },
+                            errorFields: {
+                                [CONST.QUICKBOOKS_CONFIG.AUTO_SYNC]: null,
+                            },
+                        },
+                    },
+                },
+            },
+        },
+    ];
+
+    const parameters: UpdateQuickbooksOnlineGenericTypeParams = {
+        policyID,
+        settingValue: JSON.stringify(settingValue),
+        idempotencyKey: String(CONST.QUICKBOOKS_CONFIG.AUTO_SYNC),
+    };
+    API.write(WRITE_COMMANDS.UPDATE_QUICKBOOKS_ONLINE_AUTO_SYNC, parameters, {optimisticData, failureData, successData});
+}
```

Comment:
```suggestion
function updateQuickbooksOnlineAutoSync(policyID: string, settingValue: boolean) {
    const onyxData = buildOnyxDataForQuickbooksConfiguration(policyID, CONST.QUICKBOOKS_CONFIG.AUTO_SYNC, {enabled: settingValue}, {enabled: !settingValue});
    const parameters: UpdateQuickbooksOnlineGenericTypeParams = {
        policyID,
        settingValue: JSON.stringify(settingValue),
        idempotencyKey: String(CONST.QUICKBOOKS_CONFIG.AUTO_SYNC),
    };
    API.write(WRITE_COMMANDS.UPDATE_QUICKBOOKS_ONLINE_AUTO_SYNC, parameters, onyxData);
}
```

> Because it does not follow normal convention of the functions.

@shubham1206agra are you referring to the "settingValue" being nested in `autoSync.enabled`? what about what I did above? would that work?

Reply:
this makes sense to me!